cdbcli/lexer.py

```diff
@@ -10,3 +12,36 @@
     'operand': SimpleLexer(Token.Operand),
     'database_name': SimpleLexer(Token.Operand),
 })
+
+
+def split_cli_command_and_shell_commands(command_text):
+    """Split the command text into cli commands and pipes
+
+    e.g.::
+
+        cat ID | grep text
+
+    ``cat ID`` is a CLI command, which will be matched against the CLI grammar tree
+    ``grep text`` is treated as a SHELL command
+
+    Returns a tuple ``(a, b)`` where ``a`` is the CLI command, and ``b`` is a list of shell commands
+    """
+    tokens = shlex.split(command_text)
+    parts = []
+    stack = []
+    for token in tokens:
+        # TODO: this is probably not the best way to parse shell commands
```

Comment:
Also [this](http://stackoverflow.com/questions/691148/pythonic-way-to-implement-a-tokenizer) comes to mind as well.